core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -140,32 +141,41 @@ public DockerClient client() {
                 "  Operating System: " + dockerInfo.getOperatingSystem() + "\n" +
                 "  Total Memory: " + dockerInfo.getMemTotal() / (1024 * 1024) + " MB");
 
-        String ryukContainerId = null;
         boolean useRyuk = !Boolean.parseBoolean(System.getenv("TESTCONTAINERS_RYUK_DISABLED"));
-        if (useRyuk) {
+        if (useRyuk && ryukContainerId == null) {
             ryukContainerId = ResourceReaper.start(hostIpAddress, client);
             log.info("Ryuk started - will monitor and terminate Testcontainers containers on JVM exit");
         }
 
         boolean checksEnabled = !TestcontainersConfiguration.getInstance().isDisableChecks();
         if (checksEnabled) {
-            VisibleAssertions.info("Checking the system...");
-            checkDockerVersion(version.getVersion());
-            if (ryukContainerId != null) {
-                checkDiskSpace(client, ryukContainerId);
-            } else {
-                runInsideDocker(
-                    client,
-                    createContainerCmd -> {
-                        createContainerCmd.withName("testcontainers-checks-" + SESSION_ID);
-                        createContainerCmd.getHostConfig().withAutoRemove(true);
-                        createContainerCmd.withCmd("tail", "-f", "/dev/null");
-                    },
-                    (__, containerId) -> {
-                        checkDiskSpace(client, containerId);
-                        return "";
-                    }
-                );
+            // fail-fast if checks have failed previously
+            if (cachedChecksFailure != null) {
+                throw cachedChecksFailure;
+            }
```

Comment:
It seems like a bug that currently checks can fail one time and then even try to run the checks again. We shouldn't; we should be forcing a failure so that the check failure gets investigated (rather than just the first test failing).